connect-inject/health_check_resource_test.go

```diff
@@ -0,0 +1,551 @@
+		ConsulUrl:           consulUrl,
+		ReconcilePeriod:     0,
+	}
+	return s, client, &healthResource
+}
+
+func registerHealthCheck(t *testing.T, client *api.Client, initialState string) error {
+	require := require.New(t)
+	err := client.Agent().CheckRegister(&api.AgentCheckRegistration{
+		Name:      "K8s health check",
+		ID:        testHealthCheckID,
+		ServiceID: testServiceNameReg,
+		AgentServiceCheck: api.AgentServiceCheck{
+			TTL:    "100000h",
+			Status: initialState,
+		},
+	})
+	require.NoError(err)
+	return nil
+}
+
+// We expect to already be pointed at the correct agent
+func testGetConsulAgentChecks(t *testing.T, client *api.Client) *api.AgentCheck {
+	require := require.New(t)
+	filter := fmt.Sprintf("CheckID == `%s`", testHealthCheckID)
+	checks, err := client.Agent().ChecksWithFilter(filter)
+	require.NoError(err)
+	return checks[testHealthCheckID]
+}
+
+func TestHealthCheckHandlerReconcile(t *testing.T) {
+	cases := []struct {
+		Name              string
+		CreateHealthCheck bool
+		InitialState      string
+		Pod               *corev1.Pod
+		Expected          *api.AgentCheck
+		Err               string
+	}{
+		{
+			"Reconcile new Object Create Passing",
+			false,
+			"",
+			&corev1.Pod{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      testPodName,
+					Namespace: "default",
+					Labels:    map[string]string{labelInject: "true"},
+					Annotations: map[string]string{
+						annotationStatus:  "injected",
+						annotationService: testServiceNameAnnotation,
+					},
+				},
+				Spec: corev1.PodSpec{
+					Containers: []corev1.Container{
+						corev1.Container{
+							Name: testPodName,
+						},
+					},
+				},
+				Status: corev1.PodStatus{
+					HostIP: "127.0.0.1",
+					Phase:  corev1.PodRunning,
+					Conditions: []corev1.PodCondition{{
+						Type:   corev1.PodReady,
+						Status: corev1.ConditionTrue,
+					}},
+				},
+			},
+			&api.AgentCheck{
+				CheckID: testHealthCheckID,
+				Status:  api.HealthPassing,
+			},
+			"",
+		},
+		{
+			"Reconcile new Object Create Critical",
+			false,
+			"",
+			&corev1.Pod{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      testPodName,
+					Namespace: "default",
+					Labels:    map[string]string{labelInject: "true"},
+					Annotations: map[string]string{
+						annotationStatus:  "injected",
+						annotationService: testServiceNameAnnotation,
+					},
+				},
+				Spec: corev1.PodSpec{
+					Containers: []corev1.Container{
+						corev1.Container{
+							Name: testPodName,
+						},
+					},
+				},
+				Status: corev1.PodStatus{
+					HostIP: "127.0.0.1",
+					Phase:  corev1.PodRunning,
+					Conditions: []corev1.PodCondition{{
+						Type:   corev1.PodReady,
+						Status: corev1.ConditionFalse,
+					}},
+				},
+			},
+			&api.AgentCheck{
+				CheckID: testHealthCheckID,
+				Status:  api.HealthCritical,
+			},
+			"",
+		},
+		{
+			"Reconcile existing object from passing to failing",
+			true,
+			api.HealthPassing,
+			&corev1.Pod{
```

Comment:
You could probably parametrize just the pod status condition since that's the only thing changing between the tests.

Reply:
Agree 💯

Reply:
Fair enough!